File: web/src/domain/Norm.ts
export type Article = {
  guid: string
  title: string
  marker: string
  readonly paragraphs: Paragraph[]
}

export type Paragraph = {
  guid: string
  marker: string
  text: string
}

export interface Norm {
  readonly guid: string
  readonly articles: Article[]
  officialLongTitle: string
  officialShortTitle?: string
  officialAbbreviation?: string
  referenceNumber?: string
  publicationDate?: string
  announcementDate?: string
  citationDate?: string
  frameKeywords?: string
  providerEntity?: string
  providerDecidingBody?: string
  providerIsResolutionMajority?: boolean
  leadJurisdiction?: string
  leadUnit?: string
  participationType?: string
  participationInstitution?: string
  documentTypeName?: string
  documentNormCategory?: string
  documentTemplateName?: string
  subjectFna?: string
  subjectPreviousFna?: string
  subjectGesta?: string
  subjectBgb3?: string
  unofficialLongTitle: string
  unofficialShortTitle: string
  unofficialAbbreviation: string
  risAbbreviation: string
}

File: src/main/kotlin/de/bund/digitalservice/ris/norms/domain/entity/Norm.kt
package de.bund.digitalservice.ris.norms.domain.entity

import java.time.LocalDate
import java.util.UUID

data class Norm(
    val guid: UUID,
    val longTitle: String,
    val articles: List<Article> = listOf(),

    var officialShortTitle: String? = null,
    var officialAbbreviation: String? = null,
    var referenceNumber: String? = null,
    var publicationDate: LocalDate? = null,
    var announcementDate: LocalDate? = null,
    var citationDate: LocalDate? = null,
    var frameKeywords: String? = null,
    var authorEntity: String? = null,
    var authorDecidingBody: String? = null,
    var authorIsResolutionMajority: Boolean? = null,
    var leadJurisdiction: String? = null,
    var leadUnit: String? = null,
    var participationType: String? = null,
    var participationInstitution: String? = null,
    var documentTypeName: String? = null,
    var documentNormCategory: String? = null,
    var documentTemplateName: String? = null,
    var subjectFna: String? = null,
    var subjectPreviousFna: String? = null,
    var subjectGesta: String? = null,
    var subjectBgb3: String? = null,
    var unofficialTitle: String? = null,
    var unofficialShortTitle: String? = null,
    var unofficialAbbreviation: String? = null,
    var risAbbreviation: String? = null
)
